### PR TITLE
2811 fix os logic of two

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2811.

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit and manual tests
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/common/agent_plugins/agent_plugin_manifest.py [hunk 1]
```diff
@@ -27,6 +27,7 @@ class AgentPluginManifest(InfectionMonkeyBaseModel):
     supported_operating_systems: Tuple[OperatingSystem, ...] = (
         OperatingSystem.WINDOWS,
         OperatingSystem.LINUX,
+        OperatingSystem.ANY,
```
Comment (VakarisZ) [inline monkey/common/agent_plugins/agent_plugin_manifest.py:30]: I don't think it applies here. If it supports both windows and linux, then it should be [OperatingSystem.WINDOWS, OperatingSystem.LINUX].

File: monkey/infection_monkey/network_scanning/smb_fingerprinter.py [hunk 2]
```diff
@@ -188,10 +188,12 @@ def get_host_fingerprint(
 
                 logger.debug(f'os_version: "{os_version}", service_client: "{service_client}"')
 
-                if os_version.lower() != "unix":
+                if "windows" in os_version.lower():
```
Comment (VakarisZ) [inline monkey/infection_monkey/network_scanning/smb_fingerprinter.py:191]: Use `is_windows_os` from common

File: monkey/infection_monkey/network_scanning/smb_fingerprinter.py [hunk 3]
```diff
@@ -188,10 +188,12 @@ def get_host_fingerprint(
...
                     os_type = OperatingSystem.WINDOWS
-                else:
+                elif os_version.lower() == "unix":
                     os_type = OperatingSystem.LINUX
+                else:
+                    os_type = OperatingSystem.ANY
```
Comment (VakarisZ) [inline monkey/infection_monkey/network_scanning/smb_fingerprinter.py:196]: In this case it's unknown, not any

File: monkey/infection_monkey/utils/environment.py [hunk 4]
```diff
@@ -8,11 +8,15 @@ def get_os() -> OperatingSystem:
...
+    # platform.system can return 'Linux', 'Windows', 'Darwin', 'Java'
+    # or empty string if the value can't be determined
     system = platform.system()
     if system == "Windows":
         return OperatingSystem.WINDOWS
-    if system == "Linux":
+    elif system == "Linux":
         return OperatingSystem.LINUX
+    elif system != "":
+        return OperatingSystem.ANY
```
Comment (VakarisZ) [inline monkey/infection_monkey/utils/environment.py:19]: It's not "any", it's "unknown"

Comment (ilija-lazoroski) [reply monkey/infection_monkey/utils/environment.py:19]: It is "unkown" in case if it is ""

Comment (VakarisZ) [reply monkey/infection_monkey/utils/environment.py:19]: Ahhh, my bad, it's `!=`. Then it's unsupported and should raise an error as it did.

**Comment by VakarisZ** [comment]
TBH I don't see why we need `OperatingSystem.Any`, `OperatingSystem.None`, `OperatingSystem.All` or any of the like. All of these can be expressed as a list of specific supported operating systems.

File: monkey/common/agent_plugins/agent_plugin_manifest.py (hunk 1 above)
Comment (ilija-lazoroski) [reply monkey/common/agent_plugins/agent_plugin_manifest.py:30]: Read the logic in the issue: https://github.com/guardicore/monkey/issues/2811. I think it makes sense if it is ANY.

Comment (mssalvatore) [reply monkey/common/agent_plugins/agent_plugin_manifest.py:30]: The default should just be `ANY`

File: monkey/infection_monkey/utils/environment.py (hunk 4 above)
Comment (mssalvatore) [reply monkey/infection_monkey/utils/environment.py:19]: +1, should raise exception.

**Comment by mssalvatore** [comment]
We decided not to add `OperatingSystem.Any`. Closing this.

**Pull request closed without merging**